core/templates/pages/contributor-dashboard-page/services/translate-text.service.ts

```diff
@@ -50,36 +52,42 @@ export class StateAndContent {
     public translation: string | string[],
     public dataFormat: string,
     public contentType: string,
-    public interactionId?: string,
-    public ruleType?: string
+    // The following two properties are set to null
+    // when they have no interaction.
+    public interactionId: string | null,
```

Comment:
Ditto, what are the circumstances which lead to this? Won't every state have an interaction assigned by the time it reaches the translation console?